numpy/_array_api/_creation_functions.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import annotations
+
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ._types import (List, Optional, SupportsDLPack,
+                         SupportsBufferProtocol, Tuple, Union, array, device,
+                         dtype)
+    from collections.abc import Sequence
+from ._dtypes import _all_dtypes
+
+import numpy as np
+
+def asarray(obj: Union[float, NestedSequence[bool|int|float], SupportsDLPack, SupportsBufferProtocol], /, *, dtype: Optional[dtype] = None, device: Optional[device] = None, copy: Optional[bool] = None) -> array:
+    """
+    Array API compatible wrapper for :py:func:`np.asarray <numpy.asarray>`.
+
+    See its docstring for more information.
+    """
+    # _array_object imports in this file are inside the functions to avoid
+    # circular imports
+    from ._array_object import ndarray
+    if device is not None:
+        # Note: Device support is not yet implemented on ndarray
+        raise NotImplementedError("Device support is not yet implemented")
+    if copy is not None:
+        # Note: copy is not yet implemented in np.asarray
+        raise NotImplementedError("The copy keyword argument to asarray is not yet implemented")
```

Comment:
"Maybe copy" is copy=None which is allowed here. Is your idea for guaranteed copy that we can just make a copy of the array? That would result in copying the array twice if np.asarray already does a copy.

Reply:
`asarray` supports `copy=True`? (through `np.array`)

Reply:
Ah I see. Is there any functional difference between the array() constructor and asarray? I thought maybe they were different because only one of the docstrings mention `__array__`, but I'm not sure if that's actually the case.